File: Manuel/2025-08-05/수 정렬하기 3/계수 정렬법/10989.py
import sys
import time

def main():
    input = sys.stdin.readline

    start_time = time.perf_counter()

    N = int(input()) # 수의 개수 입력
    
    # 입력 값을 모두 읽어서 저장
    input_numbers = [int(input()) for _ in range(N)]

    start_time = time.perf_counter()

    count_array = [0] * 10001

    for num in input_numbers:
        # 해당 숫자의 등장 횟수를 증가시킴
        count_array[num] += 1
    
    # count_array를 순회하며 정렬된 결과 출력
    for i in range(1, 10001):
        # 해당 숫자가 존재한다면:
        if count_array[i] > 0:
            # 등장 횟수만큼 숫자 출력
            for _ in range(count_array[i]):
                print(i)
    
    end_time = time.perf_counter()
    
    print(f"\n실행 시간: {(end_time - start_time) * 1000:.2f}ms\n", file=sys.stderr)

main()

File: Manuel/2025-08-05/수 정렬하기 3/계수 정렬법/10989.cpp
#include <iostream>
#include <vector>
#include <chrono>
#include <iomanip>
#define endl "\n"

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin >> N;
    // 입력 값 저장을 위한 임시 벡터
    vector<int> input_numbers(N);
    for(int i = 0; i < N; i++)
    {
        cin >> input_numbers[i];
    }

    auto start_time = chrono::high_resolution_clock::now();
    
    // 수의 최대 범위는 10000
    vector<int> count_array(10001, 0);
    // 입력 값의 출현 횟수를 셈
    for(int num : input_numbers)
    {
        count_array[num]++;
    }
    // count_array를 순회하며 정렬된 결과 출력
    for(int i = 1; i <= 10000; i++)
    {
        if(count_array[i] > 0)
        {
            for(int j = 0; j < count_array[i]; j++)
            {
                cout << i << endl;
            }
        }
    }

    auto end_time = chrono::high_resolution_clock::now();
    chrono::duration<double, ::milli> duration = end_time - start_time;
    cout << "\n실행 시간: " << fixed << setprecision(6) << duration.count() << "ms" << endl;
    return 0;
}